app/src/main/java/com/example/development/sakaiclientandroid/utils/custom/GradeItemAdapter.java

```diff
@@ -16,34 +19,71 @@
 
 public class GradeItemAdapter extends ArrayAdapter {
 
+    private static final float PADDING_LEFT_DP = 10f;
+    private static final float PADDING_VERTICAL_DP = 8f;
+
+
     private final Context context;
```

Comment:
@akench  It's bad practice to keep references to `Context`s because it causes severe memory leaks in some situations. Find a workaround to keeping this reference (if it's not being used just remove it).

Reply:
When would it cause memory leaks?

Reply:
The usual case for this causing leaks is when the application tries going to the background but is unable to free resources that the `Context` itself keeps reference to. This way, leaks are two-fold: You keep reference to a `Context` (which itself is a heavy object), and you keep reference to a lot of what the `Context` needs (I'm not aware how significant these are).